tests/pass/move-data-across-await-point.rs

```diff
@@ -0,0 +1,78 @@
+use std::future::Future;
+
+// This test:
+// - Compares addresses of non-Copy data before and after moving it
+// - Writes to the pointer after it has moved across the await point
+async fn data_moved_async() {
+    // Vec<T> is not Copy
+    let mut x: Vec<u8> = vec![2];
+    let raw_pointer = &mut x as *mut Vec<u8>;
+    helper_async(x, raw_pointer).await;
+    unsafe {
+        assert_eq!(*raw_pointer, vec![3]);
+        // Drop to prevent leak.
+        std::ptr::drop_in_place(raw_pointer);
+    }
+}
+
+async fn helper_async(mut data: Vec<u8>, raw_pointer: *mut Vec<u8>) {
+    let raw_pointer2 = &mut data as *mut Vec<u8>;
+    // Addresses are different because the generator upvar for `data` is a copy.
+    // To be more precise, there is a `move` happening in the MIR of the
+    // generator closure, which copies the pointer.
```

Comment:
"the pointer"? It copies the `Vec`, right?

Reply:
What I was trying to explain is that the ptr+capacity+len that is stored on the stack for that Vec is copied to the generator upvar so that's why `raw_pointer2` has a different address than `raw_pointer`. I'll see if there's a better way to phrase it in the comment